content/issues/1/weaving-interface/annotated-zoom.js

```diff
@@ -0,0 +1,72 @@
+/// todos
+// - disable scroll when positioning contextual note
+// - position contextual note better vertically
+// - disable back link in notes? (doesn't work now / not sure how to make work)
+// - fix or remove broken images
+// - link to books on derrida's margins site?
+
+
+window.addEventListener('DOMContentLoaded', (event) => {
+
+    // load annotation overlays and then use to initialize viewer
+    fetch('annotations.json')
+      .then(response => response.json())
+      .then(data => initAnnotatedZoom(data));
+
+    function initAnnotatedZoom(data) {
+
+        // create elements with page numbers to be positioned as overlays
+        var pageDiv = document.getElementById('page_numbers');
+        data.overlays.filter(el => el.className == 'page_n')
+            .forEach(el => {
+                const pageSpan = document.createElement('span');
+                pageSpan.innerText = el.content;
+                pageSpan.setAttribute('id', el.id);
+                pageDiv.appendChild(pageSpan);
+            });
```

Comment:
likely doesn't matter when there are so few elements to be added, but i think it might be slightly more performant to batch everything by creating all the spans beforehand as a string and then setting `pageDiv.innerHTML` or similar.

Reply:
Good to know — this page is a little slow so it could help. I think I'd forgotten you could do that. Do you have an example/reference handy?

Reply:
it's something i unfortunately read about _after_ having built stuff like our s&co search, which takes the same approach you used above. lemme see if i can dig up the thing I read

Reply:
so `.innerHTML` does work, but I think the approach I was actually thinking of was [this one detailed in MDN documentation](https://developer.mozilla.org/en-US/docs/Mozilla/Firefox/Performance_best_practices_for_Firefox_fe_engineers#Adding_nodes_using_DocumentFragments), which could be a pretty simple change to what you have. basically it just uses `DocumentFragment`, a node type that's designed to be more performant for use cases like this. their example maps fairly well here, i think.

Reply:
Good to know about document fragment! And easy to adapt to use it here.